nixos/modules/services/torrent/transmission.nix

```diff
@@ -191,6 +189,10 @@ in
           and [](#opt-services.transmission.settings.watch-dir).
           Note that you may also want to change
           [](#opt-services.transmission.settings.umask).
+
+          If `null`, the home and the download directories become
```

Comment:
Hmm, I think you've not understood what I tried to say previously, when I wrote:

> Besides, if cfg.downloadDirPermissions is null (the default) then StateDirectoryMode= [would default to 755 instead of the current 750](https://www.freedesktop.org/software/systemd/man/latest/systemd.exec.html#RuntimeDirectoryMode=) which would open transmission's state folder read-only to every users.

I was describing what would happen **if** we were to use `StateDirectoryMode = cfg.downloadDirPermissions;` but that's no longer the case, so `downloadDirPermissions == null` remains a no-op.

Indeed, the `description` only documents the `If not null,` case, but adding users to `config.transmission.group` is unrelated to that option being `null` or not, it's related to `StateDirectoryMode == "750"`.

Reply:
Then that option is very misleading. If I set it to 755, then I expect other users to be able to read the download directories, which is not the case. This is very unintuitive, and the option descriptions don't mention this anywhere. It took a long while before I figured out what was going on.

It would be nice if we could avoid wasting others' time. What do you suggest?

Reply:
IMHO, `downloadDirPermissions` is a quirk that should not even exist in the first place.
Yet it's there and we have to deal with it, I would suggest to improve its `description` by explaining the two main ways to give access to the directories that I explained previously:

> the better way security-wise is to add users to services.transmission.group, or to set cfg.settings.download-dir & co. to some path where intended users have access to them